lib/matplotlib/patches.py

```diff
@@ -360,17 +371,15 @@ def set_antialiased(self, aa):
         self.stale = True
 
     def _set_edgecolor(self, color):
-        set_hatch_color = True
         if color is None:
             if (mpl.rcParams['patch.force_edgecolor'] or
                     not self._fill or self._edge_default):
                 color = mpl.rcParams['patch.edgecolor']
             else:
                 color = 'none'
-                set_hatch_color = False
 
         self._edgecolor = colors.to_rgba(color, self._alpha)
-        if set_hatch_color:
+        if self._hatch_color == 'edge':
             self._hatch_color = self._edgecolor
```

Comment:
I believe we do not need these two lines anymore since `_hatch_color` stays 'edge' and is only resolved in `get_hatch_color()` to whatever edgecolor is at that time.